src/contexts/WalletContext.tsx

```diff
@@ -32,15 +34,30 @@ if (typeof window !== "undefined") {
 }
 
 export function WalletProvider(props: React.PropsWithChildren) {
+
   const [provider, setProvider] = useState<providers.Web3Provider>();
   const [walletAddress, setWalletAddress] = useState("");
+  const [walletBalance, setWalletBalance] = useState("");
+
   const getWeb3Provider = useCallback(async () => {
     const wallet = await web3Modal.connect();
     const provider = new providers.Web3Provider(wallet);
-    setWalletAddress(await provider.getSigner().getAddress());
+
+    const signer = provider.getSigner();
+    const walletAddress = await signer.getAddress()
+    const walletBalance =  await signer.getBalance() || 0
+    const formattedBalance = formatBalance(walletBalance)
+    setWalletAddress(walletAddress);
+    setWalletBalance(formattedBalance);
     setProvider(provider);
     return provider;
-  }, []);
+  }, []); 
+
+  const formatBalance = (balance:BigNumber)=>{
+    const etherFormat = utils.formatEther(balance)
+    const formatedBalance = Number(etherFormat).toFixed(4)
+    return String(formatedBalance);
```

Comment:
I think no conversion is needed, `formatedBalance` since toFixed returns `String` already.
<img width="377" alt="image" src="https://user-images.githubusercontent.com/51229945/187419167-2e350d4c-51e7-4e0d-af7d-f64b01923179.png">

Reply:
`toFixed(4)` is for rounding the balance to 4 decimal place

Reply:
Sorry!, I mean String(formatedBalance) is not needed, it is already string

Reply:
Oh I see let me correct that. Thanks for pointing that out